finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Parameter.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.BufferWriter
+import language.implicitConversions
+
+trait Parameter {
+  type A
+  def value: A
+  def evidence: CanBeParameter[A]
+
+  final def writeTo(writer: BufferWriter): Unit = {
+    evidence.write(writer, value)
+  }
+
+  final def size = evidence.sizeOf(value)
+  final def typeCode = evidence.typeCode(value)
+}
+
+object Parameter {
+  implicit def wrap[_A](_value: _A)(implicit _evidence: CanBeParameter[_A]): Parameter = {
```

Comment:
I'm nitpicking here, but maybe we could go with the `A0`, `value0` convention since `_` is already so massively overloaded in Scala? I'm not sure there's precedent for either convention in Finagle right now.